### PR TITLE
[super-agent] [FB] chore!: remove default sub-agent

### PR DESCRIPTION
**Pull request by gsanchezgavier**
#### What this PR does / why we need it:
- Adds rbac to handle secrets
- Remove default sub-agent
- Simplify rbac

#### Special notes for your reviewer:
Look at checklist before merge!!!

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] AC has been released and updated in this PR with the latest nrdot agent type
- [x] Chart Version bumped

### CODE REVIEW

File: charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl [hunk 1]
```diff
@@ -25,11 +25,9 @@ Test that the value of `.Values.config.subAgents` exists and its valid. If empty
 open-telemetry:
   type: newrelic/io.opentelemetry.collector:0.2.0
   content:
+    {{- if include "newrelic.common.nrStaging" . }}
     chart_values:
       global:
-        licenseKey: ${nr-env:NR_LICENSE_KEY}
```
Comment (paologallinaharbur) [inline charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl:30]: before merging still waiting for:
> Remove nrdot by default when installing AC chart

right?

File: charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl
Comment (paologallinaharbur) [inline charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl:1]: missing version bump

File: charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl [hunk 2]
```diff
@@ -20,17 +19,6 @@ Test that the value of `.Values.config.subAgents` exists and its valid. If empty
...
 {{- $agents | toYaml -}}
-{{- else -}}
-{{- /* Default agents for Kubernetes */ -}}
-open-telemetry:
-  type: newrelic/io.opentelemetry.collector:0.2.0
-  content:
-    {{- if include "newrelic.common.nrStaging" . }}
-    chart_values:
-      global:
-        nrStaging: true
```
Comment (paologallinaharbur) [inline charts/super-agent/charts/super-agent-deployment/templates/_helpers.tpl:33]: any docs should change to reflect that this is missing now?

**Comment by paologallinaharbur** [comment]
[ ] Guided install have been released with nrdot as default @josemore

I do not think this is a blocker since we are pinning the chart version in the UI
https://source.datanerd.us/nr1-dev-experience/shared-component-framework-configs/blob/c7cfed93c19fc1a7131d01d2d663f77fde6bbb7d/shared-component/configs/data-sources/super-agent/config.yml#L973

File: charts/super-agent/values.yaml [hunk 3]
```diff
@@ -118,22 +118,25 @@ super-agent-deployment:
...
+    #     # -- Version of the newrelic/nr-k8s-otel-collector Helm Chart.
+    #     chart_version: "0.8.0"
+    #     # -- Values to be passed to the newrelic/nr-k8s-otel-collector Helm Chart.
+    #     # By default the Agent type already has the `licenseKey` and `cluster` values set to the ones provided,
+    #     # so it is not necessary to set them here.
     #     chart_values:
-    #       global:
+    #       nr-k8s-otel-collector:
     #         licenseKey: ${nr-env:NR_LICENSE_KEY}
     #         cluster: ${nr-env:NR_CLUSTER_NAME}
```
Comment (paologallinaharbur) [inline charts/super-agent/values.yaml:138]: isn't this injected automatically now?

**Review by paologallinaharbur — DISMISSED** [review]
just a minor comments regarding the comment in the values.yaml

File: charts/super-agent/values.yaml (hunk 3 above)
Comment (gsanchezgavier) [reply charts/super-agent/values.yaml:138]: yes actually this example doesn't make much sense anymore, i will change it

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**